packages/core/content-manager/server/src/middlewares/routing.ts

```diff
@@ -1,17 +1,28 @@
-import { isNil } from 'lodash';
+import { UID, Common, Schema } from '@strapi/types';
+import type { Context, Next } from 'koa';
+import isNil from 'lodash/isNil';
 
-export default async (ctx: any, next: any) => {
-  const { model } = ctx.params;
+interface ContentType extends Schema.ContentType {
+  plugin?: string;
+}
 
-  const ct = strapi.contentTypes[model] as any;
+export default async (ctx: Context, next: Next) => {
+  const { model }: { model: UID.ContentType } = ctx.params;
+
+  const ct: ContentType = strapi.contentTypes[model];
 
   if (!ct) {
     return ctx.send({ error: 'contentType.notFound' }, 404);
   }
 
-  const target = ct.plugin === 'admin' ? strapi.admin : strapi.plugin(ct.plugin);
+  let target;
+  if (!ct.plugin || ct.plugin === 'admin') {
```

Comment:
similar to the other question, is this because of type checking?

Reply:
Yes, without this we get a TS error at L22 about the input to `strapi.plugin`